server/game/core/ability/abilityTargets/PlayerTargetResolver.ts

```diff
@@ -26,58 +28,65 @@ export class PlayerTargetResolver extends TargetResolver<IPlayerTargetResolver<A
     }
 
     protected override getAllLegalTargets(context: AbilityContext): Player[] {
         // Placeholder.
         return context.game.getPlayers();
     }
 
-    protected override resolve(context: AbilityContext, targetResults, passPrompt = null) {
-        if (!super.resolve(context, targetResults, passPrompt)) {
-            return false;
-        }
+    protected override resolveInner(context: AbilityContext, targetResults, passPrompt, player: Player, promptProperties) {
+        promptProperties.choices = ['You', 'Opponent'];
 
-        const player = context.choosingPlayerOverride || this.getChoosingPlayer(context);
-        const promptTitle = this.properties.activePromptTitle || 'Choose a player';
-        const choices = ['You', 'Opponent'];
-        const handlers = [player, player.opponent].map(
-            (chosenPlayer) => {
-                return () => {
-                    context.targets[this.name] = chosenPlayer;
-                    if (this.name === 'target') {
-                        context.target = chosenPlayer;
+        if (this.properties.mode === TargetMode.MultiplePlayers) { // Uses a HandlerMenuMultipleSelectionPrompt: handler takes an array of chosen items
+            promptProperties.activePromptTitle = promptProperties.activePromptTitle || 'Choose any number of players';
+            promptProperties.multiSelect = true;
+            promptProperties.handler = (chosen) => {
+                chosen = chosen.map((choiceTitle) => {
+                    switch (choiceTitle) {
+                        case 'You':
+                            return player;
+                        case 'Opponent':
+                            return player.opponent;
+                        default:
+                            Contract.fail('Player other than "You" or "Opponent" chosen');
                     }
-                };
-            }
-        );
-        if (player !== context.player.opponent && context.stage === Stage.PreTarget) {
+                });
+                context.targets[this.name] = chosen;
+                if (this.name === 'target') {
+                    context.target = chosen;
+                }
+                return true;
+            };
+        } else { // Uses a HandlerMenuPrompt: each choice gets its own handler, called right away when that choice is clicked
+            promptProperties.activePromptTitle = this.properties.activePromptTitle || 'Choose a player';
+            promptProperties.handlers = [player, player.opponent].map(
+                (chosenPlayer) => {
+                    return () => {
+                        context.targets[this.name] = chosenPlayer;
+                        if (this.name === 'target') {
+                            context.target = chosenPlayer;
+                        }
+                    };
+                }
+            );
+
+        /* if (player !== context.player.opponent && context.stage === Stage.PreTarget) {
```

Comment:
Just leaving a reminder to remove this if we can't sort out the issue in chat discussion